src/transformers/tokenization_gpt2.py

```diff
@@ -149,6 +149,9 @@ def __init__(
         add_prefix_space=False,
         **kwargs
     ):
+        bos_token = AddedToken(bos_token, lstrip=False, rstrip=False) if isinstance(bos_token, str) else bos_token
+        eos_token = AddedToken(eos_token, lstrip=False, rstrip=False) if isinstance(eos_token, str) else eos_token
+        unk_token = AddedToken(unk_token, lstrip=False, rstrip=False) if isinstance(unk_token, str) else unk_token
```

Comment:
This is in order to setup an unknown token if need be? It shouldn't be set by default, right, given it's a byte-level BPE?

Reply:
A little bit of background (will copy this in the description):
We now align the behavior of the byte-level BPE tokenizer to the Fast version, i.e. except for the mask token which is assumed to represent a word and thus have a prefix space, all the other are assumed to not have a prefix space.

This is already built-in for the Fast tokenizer. Here I update the slow tokenizer to have this behavior using the newly introduced `AddedToken` which lets you control the space behaviors of the special tokens.

The unk token for GPT2 is a bit strange indeed and basically here only for our tests (all the tokens are known for GPT2...) so I give him this behavior just for consistency.

Reply:
That's very cool